crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -485,6 +508,17 @@ impl WritebackCache {
         &self,
         object_id: &ObjectID,
     ) -> CacheResult<(SequenceNumber, ObjectEntry)> {
+        if let Some(entry) = self.cached.object_by_id_cache.get(object_id) {
+            let entry = entry.lock();
+            let (latest_version, latest_object) = &*entry;
+            if latest_object.is_live() {
```

Comment:
Ok, reading the PR again, I think I understand what `is_live` means, which basically try to accomodate `cache_object_not_found`. I think the criteria should be `latest_version == 0 && entry == ObjectEntry::Deleted`, right?

Or can we create an ObjectEntry::NotExist?